tests/flavor_test.go

```diff
@@ -353,6 +357,147 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
+			Expect(err).ToNot(HaveOccurred())
+
+			stashedFlavorSpec := flavorv1alpha1.VirtualMachineFlavorSpec{}
+			json.Unmarshal(flavorRevision.Data.Raw, &stashedFlavorSpec)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(stashedFlavorSpec).To(Equal(flavor.Spec))
+
+			preferenceRevision, err = virtClient.AppsV1().ControllerRevisions(util.NamespaceTestDefault).Get(context.Background(), expectedPreferenceRevisionName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			stashedPreferenceSpec := flavorv1alpha1.VirtualMachinePreferenceSpec{}
+			json.Unmarshal(preferenceRevision.Data.Raw, &stashedPreferenceSpec)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(stashedPreferenceSpec).To(Equal(preference.Spec))
+
+			By("Checking that a VirtualMachineInstance has been created with the VirtualMachineFlavor and VirtualMachinePreference applied")
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Get(vm.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vmi.Spec.Domain.CPU.Sockets).To(Equal(originalFlavorCPUGuest))
+
+			By("Updating the VirtualMachineFlavor vCPU count")
+			newFlavorCPUGuest := originalFlavorCPUGuest + 1
+			flavor.Spec.CPU.Guest = newFlavorCPUGuest
+			updatedFlavor, err := virtClient.VirtualMachineFlavor(util.NamespaceTestDefault).Update(context.Background(), flavor, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(updatedFlavor.Spec.CPU.Guest).To(Equal(newFlavorCPUGuest))
+
+			vm = tests.StopVirtualMachine(vm)
+			vm = tests.StartVirtualMachine(vm)
+
+			By("Checking that a VirtualMachineInstance has been created with the original VirtualMachineFlavor and VirtualMachinePreference applied")
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Get(vm.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vmi.Spec.Domain.CPU.Sockets).To(Equal(originalFlavorCPUGuest))
+
+			By("Creating a second VirtualMachine using the now updated VirtualMachineFlavor and original VirtualMachinePreference")
+			newVMI := tests.NewRandomVMI()
+			removeResourcesAndPreferencesFromVMI(newVMI)
+			newVM := tests.NewRandomVirtualMachine(newVMI, false)
+			newVM.Spec.Flavor = &v1.FlavorMatcher{
+				Name: flavor.Name,
+				Kind: flavorapi.SingularResourceName,
+			}
+			newVM.Spec.Preference = &v1.PreferenceMatcher{
+				Name: preference.Name,
+				Kind: flavorapi.SingularPreferenceResourceName,
+			}
+			newVM, err = virtClient.VirtualMachine(util.NamespaceTestDefault).Create(newVM)
+			Expect(err).ToNot(HaveOccurred())
+
+			newVM = tests.StartVirtualMachine(newVM)
+
+			By("Waiting for a VirtualMachineFlavorSpec ControllerRevision to be referenced from the new VirtualMachine")
+			Eventually(func() bool {
+				newVM, err = virtClient.VirtualMachine(newVM.Namespace).Get(newVM.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
Similar question as in the other `Eventually()`

Reply:
ACK. Done thanks!